PerimeterXModule/Internals/CookieDecoder.cs

```diff
@@ -19,23 +19,25 @@ public string Decode(string cookie)
 		}
 	}
 
-	public class EncryptedCookieDecoder : ICookieDecoder
-	{
-		private byte[] cookieKey;
-		private const int KEY_SIZE_BITS = 256;
+    public class EncryptedCookieDecoder : ICookieDecoder
+    {
+        private PXConfigurationWrapper pxConfig;
+
+        private const int KEY_SIZE_BITS = 256;
 		private const int IV_SIZE_BITS = 128;
 
-		public EncryptedCookieDecoder(byte[] cookieKey)
+        public EncryptedCookieDecoder(PXConfigurationWrapper pxConfig)
 		{
-			if (cookieKey == null)
+			if (pxConfig == null)
 			{
-				throw new NullReferenceException("cookieKey");
+				throw new NullReferenceException("PXConfigurationWrapper");
 			}
-			this.cookieKey = cookieKey;
+            this.pxConfig = pxConfig;
 		}
 
 		public string Decode(string cookie)
 		{
+            var cookieKeyBytes = Encoding.UTF8.GetBytes(pxConfig.CookieKey);
```

Comment:
Its because before it was receiving the cookie key in bytes on the constructor and it called it cookieKey 
`public EncryptedCookieDecoder(byte[] cookieKey)`
which  was actually this code
`var cookieKeyBytes = Encoding.UTF8.GetBytes(pxConfig.CookieKey);`